packages/devtools_shared/lib/src/memory/memory_json.dart

```diff
@@ -25,11 +25,11 @@ abstract class MemoryJson<T> implements DecodeEncode<T> {
   /// List of HeapSample.
   MemoryJson.decode(
     String payloadName, {
-    String argJsonString,
-    Map<String, dynamic> argDecodedMap,
-  }) {
+    String? argJsonString,
+    Map<String, dynamic>? argDecodedMap,
+  }) : assert(argJsonString != null || argDecodedMap != null) {
```

Comment:
Can we make named arg required?

Reply:
should both argJsonString and argDecodedMap be required? or only argJsonString?

Reply:
discussed offline - added a todo for @terrylucas